Neuro/Impostor/ImpostorHandler.cs

```diff
@@ -0,0 +1,243 @@
+    public List<Vent> NearbyVents { get; set; } = new List<Vent>();
+    // TODO: move this + related logic to Vision if necessary
+    public Vector2 DirectionToNearestVent { get; set; } = Vector2.zero;
+    public List<PlainDoor> NearbyDoors { get; set; } = new List<PlainDoor>();
+
+    // member variables used in example methods
+    public bool goingForKill { get; set; } = false;
+    public PlayerControl killTarget { get; set; } = null;
+
+    private void FixedUpdate()
+    {
+        if (!ShipStatus.Instance) return;
+        if (MeetingHud.Instance) return;
+        if (!PlayerControl.LocalPlayer) return;
+
+        // TODO: maybe try and make this only update once
+        isImpostor = PlayerControl.LocalPlayer.Data.RoleType == AmongUs.GameOptions.RoleTypes.Impostor;
+
+        // TODO: Move UpdateNearbyDoors when implementing maps with doors that non-impostors can use
+        if (isImpostor)
+        {
+            UpdateNearbyVents();
+            UpdateNearbyDoors();
+            // GetOrKillTarget();
+            // AttemptVent();
+        }
+    }
+
+    [HideFromIl2Cpp]
+    private void UpdateNearbyVents()
+    {
+        Vent closest = null;
+        float closestDistance = 999f;
+        NearbyVents.Clear();
+        foreach (Vent vent in ShipStatus.Instance.AllVents)
+        {
+            float distance = Vector2.Distance(vent.transform.position, PlayerControl.LocalPlayer.transform.position);
+            if (distance < 10f)
+            {
+                NearbyVents.Add(vent);
+                // also take the opportunity to get the closest vent
+                if (distance < closestDistance)
+                {
+                    closest = vent;
+                    closestDistance = distance;
+                }
+            }
+        }
+        ClosestVent = closest;
+
+        if (ClosestVent == null)
+        {
+            DirectionToNearestVent = Vector2.zero;
+        }
+        else
+        {
+            DirectionToNearestVent = (ClosestVent.transform.position - PlayerControl.LocalPlayer.transform.position).normalized;
+        }
+        
+    }
+
+    [HideFromIl2Cpp]
+    private void UpdateNearbyDoors()
+    {
+        NearbyDoors.Clear();
+        foreach (PlainDoor door in ShipStatus.Instance.AllDoors)
+        {
+            float distance = Vector2.Distance(door.transform.position, PlayerControl.LocalPlayer.transform.position);
+            if (distance < 10f)
+            {
+                NearbyDoors.Add(door);
+            }
+        }
+    }
+
+    /** ----- Stubs below here ----- **/
+
+    // example of kill functionality
+    private void GetOrKillTarget()
+    {
+        if (PlayerControl.LocalPlayer.killTimer > 0f) return;
+
+        if (killTarget == null && !PlayerControl.LocalPlayer.inVent)
+        {
+            PlayerControl potentialKillTarget = null;
+
+            foreach (KeyValuePair<PlayerControl, LastSeenPlayer> player in NeuroPlugin.Instance.Vision.GetPlayerLocations())
```

Comment:
Use deconstructor here
```suggestion
            foreach ((PlayerControl player, LastSeenPlayer lastSeen) in NeuroPlugin.Instance.Vision.GetPlayerLocations())
```